Core.DomainModel/GDPR/DataProcessingRegistration.cs

```diff
@@ -340,6 +340,48 @@ public Maybe<IEnumerable<DataProcessingRegistrationOversightDate>> SetOversightC
         }
 
         public virtual ICollection<ItContract.ItContract> AssociatedContracts { get; set; }
+        public int? MainContractId { get; set; }
+        public virtual ItContract.ItContract MainContract { get; set; }
+        public bool IsActiveAccordingToMainContract => CheckContractValidity();
+
+        public void ResetMainContract()
+        {
+            MainContract?.Track();
+            MainContract = null;
+        }
+
+        public Maybe<OperationError> AssignMainContract(int contractId)
+        {
+            if (MainContract != null && contractId == MainContract.Id)
+                return Maybe<OperationError>.None;
+
+            var contract = AssociatedContracts.FirstOrDefault(c => c.Id == contractId);
+            if (contract == null)
+                return new OperationError($"Contract with id: {contractId} is not associated with this data processing registration", OperationFailure.BadState);
+
+            ResetMainContract();
+            MainContract = contract;
+
+            return Maybe<OperationError>.None;
+        }
+
+        public Maybe<OperationError> RemoveMainContract(int contractId)
```

Comment:
I wanted to reuse this method for "v2 Write service" but I had a problem with using id vs uuid, as in the UI contracts (in the dpr tab) use the id. But still I'm not sure if it's a right solution to use ResetMainContract in the WriteService?